drivers/usbhid-ups.c

```diff
@@ -1422,8 +1432,22 @@ static void ups_status_set(void)
 		dstate_delinfo("input.transfer.reason");
 	}
 
-	if (ups_status & STATUS(ONLINE)) {
-		status_set("OL");		/* on line */
+
+	if ((ups_status & STATUS(ONLINE)) && (ups_status & STATUS(DISCHRG))) {
+		/* warning or CyberPower UT quirk */
+		if (cputquirk) {
+			status_set("OB");
+		} else if ((ups_status & STATUS(CAL))) {
+			status_set("OL");
+		} else {
+			upslogx(LOG_WARNING, "%s: seems that UPS [%s] is OL+DISCHRG state now. "
```

Comment:
English: ups "has" or "is in" some state

Reply:
typo, fixed